core/src/tile/tileWorker.cpp

```diff
@@ -76,7 +76,7 @@ void TileWorker::run(Worker* instance) {
                     if (a->isProxy() != b->isProxy()) {
                         return !a->isProxy();
                     }
-                    if (a->source().id() == b->source().id() &&
+                    if (a->sourceId() == b->sourceId() &&
```

Comment:
We can save on saving sourceId in tileTask, and instead modify this lambda to give priority to the worker containing a valid not-deleted source.

Reply:
Not sure what you see to improve here

Reply:
Why to check for sourceId for a source which might have been deleted?

Reply:
It's probably rare that a source is deleted here. The intention was to not lock (increment/decrement ref counters) the source in a loop generally. As there are only few items compare it would be ok in this case. We could do it either way